File: app/src/main/java/quangnam/com/sample/di/PerApplication.java
package quangnam.com.sample.di;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;

/**
 * Created by quangnam on 10/16/17.
 * Project base
 */

@Qualifier
@Retention(RetentionPolicy.RUNTIME)
public @interface PerApplication {
}

File: app/src/main/java/quangnam/com/sample/di/PerApplication.kt
package quangnam.com.sample.di

import java.lang.annotation.RetentionPolicy
import javax.inject.Scope

/**
 * Created by quangnam on 10/16/17.
 * Project base
 */

@Scope
@Retention(AnnotationRetention.RUNTIME)
annotation class PerApplication
